geopandas/tools/tests/test_sjoin.py

```diff
@@ -45,6 +45,12 @@ def dfs(request):
         df1.index.name = "df1_ix"
         df2.index.name = "df2_ix"
 
+    if request.param == "multi-index":
+        i1 = ["a", "b", "c"]
+        i2 = ["d", "e", "f"]
+        df1 = df1.set_index([i1, i2])
+        df2 = df2.set_index([i2, i1])
```

Comment:
Should we also have a named multi-index case? (do the names get preserved?)

Reply:
Yes, will do. Names get preserved.